lib/cli/src/Cardano/CLI.hs

```diff
@@ -755,14 +774,13 @@ cmdKeyInspect =
     cmd = fmap exec $
         KeyInspectArgs <$> keyArgument
 
-    exec (KeyInspectArgs key) =
-        inspect scheme key >>= TIO.putStrLn
-
-    scheme :: CliKeyScheme Text IO
-    scheme =
-        hoistKeyScheme eitherToIO
-        . mapKey keyHexCodec
-        $ newCliKeyScheme Icarus
+    exec (KeyInspectArgs key) = do
+        orReadStdin key
+            >>= eitherToIO . go
+            >>= TIO.putStrLn
+      where
+        go :: Text -> Either String Text
+        go = fmap (\(k, _) -> inspect k) . decodeAnyKey
```

Comment:
small note but `go` is typically used for recursive functions. So it feels like something is going to recurse but it isn't. Perhaps `action` or `run` is a better fit.